lib/mondrian/olap/result.rb

```diff
@@ -481,8 +478,70 @@ def self.parse_return_fields(result, params)
             end
           end
 
+          if params[:role_name].present?
+            add_role_validation_fields return_fields, sql_options
```

Comment:
Probably better `add_role_restriction_fields` as we are not validating roles but restricting results based on roles.